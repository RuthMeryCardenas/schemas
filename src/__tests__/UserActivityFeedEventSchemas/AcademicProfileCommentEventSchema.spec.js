const mongoose = require('mongoose/browser');
const UserSchema = require('../../UserSchema')(mongoose);
const { UserActivityFeedEventSchemas: { AcademicProfileCommentEventSchema } } = require('../../../')(mongoose);


describe('AcademicProfileCommentEventSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, AcademicProfileCommentEventSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const createdBy = new mongoose.Document({
      uid: 'xxxxxxxxxx',
      email: 'foo@bar.baz',
    }, UserSchema);

    const doc = new mongoose.Document({
      createdBy,
      commentType: 'tech',
      project: 'Placeholder project title',
      text: 'Comment text',
    }, AcademicProfileCommentEventSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
