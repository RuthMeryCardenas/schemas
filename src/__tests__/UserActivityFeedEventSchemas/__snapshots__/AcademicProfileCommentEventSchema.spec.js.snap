// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileCommentEventSchema should fail validation when missing fields are provided 1`] = `
Object {
  "commentType": [ValidatorError: Path \`commentType\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "project": [ValidatorError: Path \`project\` is required.],
  "text": [ValidatorError: Path \`text\` is required.],
}
`;
